File: meta_context.py
class MetaContext:
    def __init__(self, tree, line_offset, parent_type, parent_name=None):
        self.tree = tree
        self.line_offset = line_offset
        self.parent_type = parent_type
        self.parent_name = parent_name

    def num_of_lines(self):
        return self.tree.root_node.end_point[0] - 1

File: wuff/document/MetaContext.cpp
//
// Created by Michal Janecek on 30.01.2024.
//

#include "MetaContext.h"


MetaContext::MetaContext(TSTree *tree, uint32_t lineOffset, uint32_t byteOffset, const std::string &parentType,
                         const std::string &parentName)
    : tree(tree), lineOffset(lineOffset), byteOffset(byteOffset), parentType(parentType), parentName(parentName) // Initializer list
{
}
